src/librados/AioCompletionImpl.h

```diff
@@ -85,10 +85,7 @@ struct librados::AioCompletionImpl {
     return wait_for_complete();
   }
   int is_complete() {
-    lock.Lock();
-    int r = complete;
-    lock.Unlock();
-    return r;
+    return complete;
```

Comment:
Thank you for the reply. yes, just as you said, there are two types of reordering. CPU reordering and complier reordering. 
In this link we can see x86 is strong order https://en.m.wikipedia.org/wiki/Memory_ordering
About complier reordering, we can add the memory barrier to avoid this happened  “asm volatile ("" ::: "memory")”.
On x86 platform compiler memory barrier is enough. 	https://stackoverflow.com/questions/12183311/difference-in-mfence-and-asm-volatile-memory
But on other weak order CPU, We should add additional CPU barrier instruction to keep the order, such as PPC(sync), MIPS(sync), POWER(dcs) ARM(dmb,dsb,isb)
This is light weight than Mutex, it is benefit to performance.